File: ruby/raindrops/raindrops.rb
# frozen_string_literal: false

# https://exercism.org/tracks/ruby/exercises/raindrops
# Raindrops exercise
class Raindrops
  def self.convert(number)
    sounds = ''

    sounds = "#{sounds}Pling" if number.pling?
    sounds = "#{sounds}Plang" if number.plang?
    sounds = "#{sounds}Plong" if number.plong?

    sounds = number.to_s if sounds.empty?

    sounds
  end
end

# Adding sound methods to integers.
class Integer
  def pling?
    (self % 3).zero?
  end

  def plang?
    (self % 5).zero?
  end

  def plong?
    (self % 7).zero?
  end
end

File: python/raindrops/raindrops.py
"""Raindrops exercism: https://exercism.org/tracks/python/exercises/raindrops"""


def convert(number: int) -> str:
    """Converts a number to the sound of rain drops.

    :param number: positive integer
    :return: string with rain sounds or the number if it can't be converted
    """

    sounds: list[str] = []

    if number % 3 == 0:
        sounds.append("Pling")

    if number % 5 == 0:
        sounds.append("Plang")

    if number % 7 == 0:
        sounds.append("Plong")

    if len(sounds) == 0:
        sounds.append(f"{number}")

    return "".join(sounds)
